pkg/nexpose/asset.go

```diff
@@ -0,0 +1,272 @@
+package nexpose
```

Comment:
This package appears to contain only structs, is imported by `domain`, and contains JSON annotations. This skews quite a bit from our standard layout and practices. I'll try to decompose this into a few threads of though.

re: only structs. There's nothing inherently wrong with having a package of structs. However, if those structs are used to communicate across packages then they are actually part of the `domain` and should be kept there. The expectation for our work is that each project will have `domain` and `logs` as standard packages and all other packages are implementations of something from `domain`.

re: imported by domain. The `domain` package is intended to be a root of the import graph. It should never import local packages. We occasionally import 3rd party packages and either include or alias the types in `domain` in order to mask some implementation detail from the rest of the project. However, importing a local package stands out as this heavily indicates that the local package is actually part of the `domain`.

re: JSON annotations. We had a similar discussion in one of @LorenBoren PRs. Typically, we forbid JSON annotations in `domain` because this couples the `domain` to some specific implementation. For example, most projects would have a `domain.Asset` type with some set of fields and then a `nexpose/http.go` file would contain its own set of types to and from which it adapts the information from `domain` in order to communicate with the concrete implementation. These types would then contain the JSON annotations needed to communicate with the remote API. Without this separation we end up needing to make large scale refactors every time the remote API adds or removes a field.

The exception we made in another PR was that we'd tolerate having the concrete in `domain` only if the project was specifically scoped to the concrete. In the other case, the project was limited in scope to the concrete and any other implementation would be invalid for the use case. This, from what we've seen, is almost always the exception but can be true from time to time. If this project fits that description then we can keep the JSON tags for ease of use.

Reply:
Makes sense. The same exception for @LorenBoren 's PR applies here, but I will remove the custom local package in favor of `domain`.